tests/ImageSharp.Tests/Color/ColorTests.CastTo.cs

```diff
@@ -88,6 +88,19 @@ public void Bgr24()
                 Bgr24 data = color;
                 Assert.Equal(source, data);
             }
+
+            [Fact]
+            public void TPixel()
+            {
+                var source = new RgbaVector(float.Epsilon, 2 * float.Epsilon, float.MaxValue, float.MinValue);
+
+                // Act:
+                var color = Color.FromPixel(source);
+
+                // Assert:
+                RgbaVector data = color.ToPixel<RgbaVector>();
+                Assert.Equal(source, data);
+            }
```

Comment:
```suggestion
            public static TheoryData<IPixel> GenericPixel_Data = new TheoryData<IPixel>
            {
                { new new RgbaVector(float.Epsilon, 2 * float.Epsilon, float.MaxValue, float.MinValue) },
                { new Rgba64((ushort)1, (ushort)2, ushort.MaxValue, (ushort)(ushort.MaxValue - 1)) },
                { new Rgb48((ushort)1, (ushort)2, (ushort)(ushort.MaxValue - 1))) },
                { new La32((ushort)1, (ushort)(ushort.MaxValue - 1))) },
                { new L16((ushort)(ushort.MaxValue - 1))) },
                { new Rgba32(1, 2, 255, 254) },
            }

            [Theory]
            [MemberData(nameof(GenericPixel_Data))]
            public void GenericPixel<TPixel>(TPixel source)
                where TPixel : unmanaged, IPixel<TPixel>
            {
                // Act:
                var color = Color.FromPixel(source);

                // Assert:
                TPixel actual = color.ToPixel<TPixel>();
                Assert.Equal(source, actual);
            }
```

Haven't checked if it compiles ..

Reply:
This is doing my head in. Tried to hack around it but not getting very far.
https://github.com/xunit/xunit/issues/2061

Reply:
Hacked it, those implicit casts are a nightmare.